addons/account/models/account_move.py

```diff
@@ -371,17 +371,30 @@ def _get_accounting_date(self, invoice_date, has_tax):
         last date of the first open period.
 
         :param invoice_date (datetime.date): The invoice date
-        :param has_tax (bool): Iff any taxes are involved in the lines of the invoice
+        :param reverse: Iff self is the reversal of another move.
         :return (datetime.date):
         """
         tax_lock_date = self.company_id.tax_lock_date
+        period_lock_date = self.company_id.period_lock_date
         today = fields.Date.today()
-        if invoice_date and tax_lock_date and has_tax and invoice_date <= tax_lock_date:
+        has_tax = bool(self.line_ids.tax_ids or self.line_ids.tax_tag_ids)
+        tax_lock_violated = invoice_date and tax_lock_date and has_tax and invoice_date <= tax_lock_date
+        period_lock_violated = invoice_date and period_lock_date and invoice_date <= period_lock_date
+        if tax_lock_violated:
             invoice_date = tax_lock_date + timedelta(days=1)
+        if period_lock_violated:
+            invoice_date = period_lock_date + timedelta(days=1)
```

Comment:
`invoice_date = max(invoice_date, tax_lock_violated, period_lock_violated)` ?
If both dates are != and violated, this is weird you get the date after the period lock date even if the tax lock date is set after